lib/store/task.ts

```diff
@@ -1,12 +1,15 @@
 import { create } from 'zustand';
 import { persist } from 'zustand/middleware';
-import { Task } from '@/lib/db';
-import { taskService } from '@/lib/services';
+import { Task, USER_ID } from '@/lib/db';
+import { taskService, tagsService } from '@/lib/services';
+import { db } from '@/lib/db';
 
 interface TaskStore {
   tasks: Task[];
+  currentTagId: string | null;
+  setCurrentTagId: (tagId: string | null) => void;
   fetchTasks: () => Promise<void>;
-  addTask: (task: string, color?: string) => void;
+  addTask: (task: string, color?: string, taskTag?: string) => void;
```

Comment:
I don't love mixing the Tags and Tasks concerns here.... 

I suspect once I move over to a Prisma DB and have queries available this could be alleviated, and I should prioritize that move.